PocketKit/Sources/Sync/Widgets/WidgetsSessionService.swift

```diff
@@ -9,6 +9,7 @@ import WidgetKit
 public protocol WidgetsSessionService {
     var isLoggedIn: Bool { get }
     func setLoggedIn(_ isLoggedIn: Bool)
+    var widgetsDetails: [String] { get }
```

Comment:
do we need to store `widgetDetails` into `UserDefaults`, or can we just call `getWidgetConfigurations()` when we need it? This way we ensure to always obtain up to date information

Reply:
we can call it directly, but storing it in the userDefaults will allow us to keep track if a user deletes / adds a widget as a workaround

See comment: https://github.com/Pocket/pocket-ios/pull/779#discussion_r1238745988

Wanted to get your input, but can remove the userDefaults and revisit in the future.

Reply:
So yes, if we change the call site (see this [comment](https://github.com/Pocket/pocket-ios/pull/779#discussion_r1238938299)) it might be worth storing into `UserDefaults`. Not sure that `sessionService` is the best place to do it. We could add a dedicated `WidgetTracker` object (injected in the `TimelineProvider` possibly`) that takes care of that

Reply:
updated and created a `WidgetTracker`